### ISSUE TITLE
Setup Crowdin integration for release branch

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9031]
While integrating the direct Crowdin connection I left out other branches than `master` intentionally, as we had some strange effects in the past (with the script based solution).
As we however have seen in the recent past, it might be wise to be able to update strings (and thus their translations) also on `release`. As time permits I plan to look into this multi branch Crowdin integration once again. Maybe I use the WhereYouGo repo as a trial (because of less commit activity there).

I might need some help however later on to try it out by explicitly adding a new string to `release`, merging it back into `master` to see what happens on Crowdin and the corresponding PRs from Crowdin.
I will follow up here.

**Comment by bekuno** [comment]
That's a good idea.
You can trigger me when you need support.

**Comment by Lineflyer** [comment]
I will try on WhereYouGo next days (starting now).
I informed contributors via Crowdin mail.

The tests needed (as these made problems in the past) with our branch constellation are IMHO:

- [ ] After we add `release` to Crowdin it would theoretically be necessary to translate all of the strings for this branch once again (which might - additionally to the effort - also lead to diffs between `master` and `release` as strings might be translated slightly different on the branches). This might not apply, as Crowdin has some intelligent feature to take over existing translations (to be tested). If the problem still applies, its more or less a one time work to upload all existing translations of `release` to Crowdin (workaround).

- [ ] In case a new string comes in on `release` the corresponding string will (after merging to `master`as we do it) appear on both branches. If we merge translations from Crowdin for `release` and later for `master` we might get conflicts as same files are touched (although the content might be the same). Addon: If the content is the same we will have no diff but still it depends from the base where Crowdin creates it branches `l10n_*` from (meaning: does it already contain the merged translations from the other branch or not).

**Comment by Lineflyer** [comment]
@bekuno 
As you offered to help and my time is a bit limited at the moment:
What I did:
- I added branch `release` of the WhereYouGo repository on crowdin
- I initialized a force sync (which reads sources to Crowdin, creates branch `l10n_release` on Github and opens a PR for the diff)

The outcome is https://github.com/cgeo/WhereYouGo/pull/213

What needs to be analyzed/understood:
- Why do we have this diff in the mentioned PR? Is it reasonable (it can IMHO only be reasonable if the strings are present on both branches but the new translations only came in on `master`, meaning they have been contributed after `release` was last branched off from `master`)
- When I check the translations offered on Crowdin I do still see only `master` as base dir while I expected `release` to be added. But maybe Crowdin is intelligent and detects, that there is no diff in strings between those branches. I did however IMHO never tell, that `master` is the master...

**Comment by Lineflyer** [comment]
@rsudev Do you have any insights from the past "script approach" when it comes to multiple branches?

**Comment by Lineflyer** [comment]
Sorry, couldn't resists to dive deeper already:
Right now it looks really reasonable to me:
- Last merge from `master` into `release` for WhereYouGo was on August, 21st
- PR https://github.com/cgeo/WhereYouGo/pull/206 was merged after that on August, 28th (containing seemingly - not checked in detail - the changes to Hungarian and Japanese also present in the new Crodwin PR)
- PR https://github.com/cgeo/WhereYouGo/pull/211 was merged also after branching in September (containing seemingly - not checked in detail - the change in Korean also present in the new Crowdin PR)

Conclusion (@bekuno Please verify):
- Strings have been included on both branches but were only translated on `master` (as it was the only branch configured on Crowdin)
- After enabling `release` branch on Crowdin it performs better as expected and delivers the strings only translated on `master` but missing on `release` also to `release`

**Comment by bekuno** [comment]
I started a test 
- I created a string in strings.xml on the release branch
- the string was uploaded to crowdin, now the release tree with strings.xml became visible
  only changed strings are necessary to translate
- I translated/approved the string in german
- at next the commit should be generated from the cgeo-ci-bot; let's wait

Btw., there is a bug in crowdin counting the strings:
![grafik](https://user-images.githubusercontent.com/32555742/94619503-4e3ab480-02ad-11eb-8b4d-b9c20d36bc10.png)
I created only one string but an additional comment row.
After approving the single string it shows 100% finished

@Lineflyer maybe you can force the update?

**Comment by bekuno** [comment]
PR #213 was now updated. This seems fine now - files for all languages created because of the comment, in german also the string translation.

**Comment by Lineflyer** [comment]
Just leave it all open right now.
Will take a look next weekend, but sounds promising already.

**Comment by Lineflyer** [comment]
note to myself:
We should additionally test changing a string on release and merging right away to master as this is what we normally do. What PR will come from Crowdin for what branch and will there be any conflicts in PRs.

**Comment by bekuno** [comment]
> 
> 
> We should additionally test changing a string on release and merging right away to master as this is what we normally do. What PR will come from Crowdin for what branch and will there be any conflicts in PRs.

A new string at release is added. So you can start the workflow.
And afterwards the same with the PR removing the string.

**Comment by Lineflyer** [comment]
According to 

> note to myself:
We should additionally test changing a string on release and merging right away to master as this is what we normally do. What PR will come from Crowdin for what branch and will there be any conflicts in PRs.

I will now merge `release` back to `master` without prior merging translations as we normally want to merge `release` right away I will just close the related PR).
Lets see what happens then.

**Comment by Lineflyer** [comment]
Now we have PR https://github.com/cgeo/WhereYouGo/pull/215 for `master` which looks good and consistent in itself (now contains also the test string as it was merged to `master` including the already done translation in german).
Additionally Crowdin opened a new PR https://github.com/cgeo/WhereYouGo/pull/216 for `release` which contains the same changes (logical as both branches are now on same strings).

To create my conflict case 

> In case a new string comes in on `release` the corresponding string will (after merging to `master`as we do it) appear on both branches. If we merge translations from Crowdin for `release` and later for `master` we might get conflicts as same files are touched (although the content might be the same). Addon: If the content is the same we will have no diff but still it depends from the base where Crowdin creates it branches `l10n_*` from (meaning: does it already contain the merged translations from the other branch or not).

by merging the PR for `master` first and see what happens with the PR for `release`.

**Comment by Lineflyer** [comment]
Works perfect and without conflicts.

Master history looks like this now:
```
*   eaa4ab3 (HEAD -> master, origin/master, origin/HEAD) Merge branch 'release'
|\
| * 8c9fc95 (origin/release, release) New Crowdin updates (#216)
* | b756da6 New Crowdin updates (#215)
* |   fb2a35e Merge branch 'release'
|\ \
| |/
| * f3bbf87 Update strings.xml
* | 610d81d (origin/l10n_master) New translations strings.xml (Korean) (#211)
* | 584d258 New Crowdin updates (#206)
```

This means we do not have any issue I do see right now with the branch approach we are using.

I will continue with enabling it for c:geo repo after double checking tonight (using the revert PR).

**Comment by Lineflyer** [comment]
While removing the string on both branches I got a strange PR https://github.com/cgeo/WhereYouGo/pull/218 which consists of removal of the test string althouhg the removal has already been done by merging the respective PR from Crowdin on `release` and merging that result back to `master`.

I explained the reason in that PR and will add it to the WIki (Translation page).

Summary:
All is working as intended, no conflicts, no double translation effort, only some Need-to-Know.

Will continue now with c:geo repository.
